staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -944,6 +1019,24 @@ func (alloc *allocator) isSelectable(r requestIndices, requestData requestData,
 
 }
 
+// isConsumable checks whether a device with remaining resources is consumable by the request.
+func (alloc *allocator) CmpRequestOverCapacity(request requestAccessor, slice *draapi.ResourceSlice, deviceIndex int) (bool, error) {
+	deviceID := DeviceID{Driver: slice.Spec.Driver, Pool: slice.Spec.Pool.Name, Device: slice.Spec.Devices[deviceIndex].Name}
+	var allocatingCapacityPtr *ConsumedCapacity
+	if allocatingCapacity, found := alloc.allocatingCapacity[deviceID]; found {
+		allocatingCapacityPtr = &allocatingCapacity
+	}
```

Comment:
If there is no such entry in the map, the zero-value is returned - nil in this case:
```suggestion
	allocatingCapacityPtr := alloc.allocatingCapacity[deviceID]
```